src/CoreWCF.Http/tests/Helpers/ClientHelper.cs

```diff
@@ -62,6 +62,29 @@ private static void ApplyDebugTimeouts(Binding binding)
                     binding.ReceiveTimeout = s_debugTimeout;
             }
         }
+
+        public static Binding GetBasicHttpBinding()
+        {
+            BasicHttpBinding basicHttpBinding = new BasicHttpBinding();
+            basicHttpBinding.TransferMode = TransferMode.StreamedResponse;
+            basicHttpBinding.MaxReceivedMessageSize = Int32.MaxValue;
+            basicHttpBinding.SendTimeout = TimeSpan.FromMinutes(3);
+            basicHttpBinding.ReceiveTimeout = TimeSpan.FromMinutes(3);
+            return basicHttpBinding;
+        }
+
+        public static Binding GetCustomBinding()
+        {
+            HttpTransportBindingElement httpBE = new HttpTransportBindingElement();
+            httpBE.MaxReceivedMessageSize = int.MaxValue;
+            httpBE.TransferMode = TransferMode.StreamedResponse;
```

Comment:
@mconnew, with config "httpBE.TransferMode = TransferMode.StreamedResponse;"  applied in serivce, calling the CustomBinding service operation which returns a FileStream throws following exception, if remove this setting, the test would pass. Is it a bug?

> Message: 
>     System.ServiceModel.CommunicationException : The underlying connection was closed: The connection was closed unexpectedly.
>     ---- System.Net.WebException : The underlying connection was closed: The connection was closed unexpectedly.
>   Stack Trace: 
>     Server stack trace: 
>     HttpChannelUtilities.ProcessGetResponseWebException(WebException webException, HttpWebRequest request, HttpAbortReason abortReason)
>     HttpChannelRequest.WaitForReply(TimeSpan timeout)
>     RequestChannel.Request(Message message, TimeSpan timeout)
>     RequestChannelBinder.Request(Message message, TimeSpan timeout)
>     ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout)
>     ServiceChannelProxy.InvokeService(IMethodCallMessage methodCall, ProxyOperationRuntime operation)
>     ServiceChannelProxy.Invoke(IMessage message)
>     Exception rethrown at [0]: 
>     RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
>     RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
>     IService.GetStream()
>     AsyncStreamingTests.SlowStreamConsumingClient(IService serviceChannel, Int32 requests) line 130
>     AsyncStreamingTests.StreamCustomBindingTest() line 61
>     ----- Inner Stack Trace -----
>     HttpWebRequest.GetResponse()
>     HttpChannelRequest.WaitForReply(TimeSpan timeout)

Reply:
That is a bug. A Task is being completed prematurely on the server side triggering ASP.NET Core to close the port. I need to make sure that the task isn't completed before the response has finished sending. I'll leave this PR open until I've worked out the fix and commited it. Then this test should pass.